### DISCUSSION TITLE
AutoTiller connection

### DISCUSSION THREAD

**Discussion by EarlWer** [bareboat-necessities/lysmarine_gen#283] [Q&A] [answered]
I am planning to buy an Autotiller. Probably the Raymarine ST-2000. It looks like the previous owner had one installed.

Has anyone done this before?
I am assuming (you know what that means...) that the Seatalk and/or NMEA 0183 will be compatible with my RPi.

I should be able to set a waypoint in OpenCPN and have the tiller steer that course, correct?

Physical connection: Do I need any extra cables or converters? I will need to extend the Raymarine plug to the RPi, but how is it plugged into the RPi?  If a HAT is needed, which one?

thanks

**Comment by mgrouch** [comment, marked as answer]
BBN OS comes with pypilot preinstalled. 
With additional IMU and motor controller hardware (see the documentation) you will be able to control tiller pilot. 

Pypilot is preconfigured to receive waypoints data from OpenCPN and SignalK. 

Pypilot hat is optional hardware. It has small screen and support for IR Remote control. 

To input seatalk1 data you have options. Best is just one of commercial multiplexers (with USB to pi4). 
If you have seatalk device to control autopilot then multiplexer needs to support bi-directional translation on Seatalk to NMEA-0183. 

For demo of the working set up you can check this video 

https://youtu.be/3zMjUs2X3qU

**Comment by mgrouch** [comment]
You can also just use linear actuator connected to the motor controller. 

Here is the discussion thread on pypilot forums:

https://forum.openmarine.net/showthread.php?tid=3468
